src/tools/run-make-support/src/command.rs

```diff
@@ -24,18 +24,50 @@ use build_helper::drop_bomb::DropBomb;
 /// [`run`]: Self::run
 /// [`run_fail`]: Self::run_fail
 /// [`run_unchecked`]: Self::run_unchecked
-#[derive(Debug)]
 pub struct Command {
     cmd: StdCommand,
     stdin: Option<Box<[u8]>>,
+    completed_process_checks: Vec<(String, Box<dyn Fn(&CompletedProcess)>)>,
```

Comment:
> ```rust
> #[derive(Debug)] // this works now
> pub struct Command {
>     cmd: StdCommand,
>     stdin: Option<Box<[u8]>>,
>     assert_stdout_ends_with_newline: bool,
>     drop_bomb: DropBomb,
> }
> ```
> 
> and
> 
> ```rust
> // Command::command_output
> if self.assert_stdout_ends_with_newline {
>     completed_process.assert_stdout_ends_with_newline();
> }
> ```

Let's use a simple bool for now, if there are more check-hooks that we wish to register upon construction, then we can always refactor into the hook form when it comes to it.

@rustbot author